hnn_core/network_builder.py

```diff
@@ -358,26 +367,32 @@ def _gid_assign(self):
 
         self.net._update_cells()  # updates net.n_cells
 
-        rank = _get_rank()
+        rank = self._rank
         nhosts = _get_nhosts()
 
         # round robin assignment of gids
         for gid in range(rank, self.net.n_cells, nhosts):
             # set the cell gid
-            _PC.set_gid2node(gid, rank)
             self._gid_list.append(gid)
 
         # loop over all drives, then all cell types, then all artificial cells
         # only assign a "source" artificial cell to this rank if its target
         # exists in _gid_list. "Global" drives get placed on different ranks
         for drive in self.net.external_drives.values():
-            for conn in drive['conn'].values():  # all cell types
-                for src_gid, target_gid in zip(conn['src_gids'],
-                                               conn['target_gids']):
-                    if (target_gid in self._gid_list and
-                            src_gid not in self._gid_list):
-                        _PC.set_gid2node(src_gid, rank)
-                        self._gid_list.append(src_gid)
+            if drive['cell_specific']:
+                for conn in drive['conn'].values():  # all cell types
+                    for src_gid, target_gid in zip(conn['src_gids'],
+                                                   conn['target_gids']):
+                        if (target_gid in self._gid_list and
+                                src_gid not in self._gid_list):
+                            self._gid_list.append(src_gid)
+            else:
+                src_gids = list(self.net.gid_ranges[drive['name']])
+                for gid_idx in range(rank, len(src_gids), nhosts):
+                    self._gid_list.append(src_gids[gid_idx])
+
+        for gid in self._gid_list:
+            _PC.set_gid2node(gid, rank)
```

Comment:
@rythorpe see how I separated the Neuron code. Now you can test the code above without making a call to Neuron

Reply:
Ahh yes, now I see it. Thanks @jasmainak!